workers/unity/Assets/Gdk/Core/Systems/CleanReactiveComponentsSystem.cs

```diff
@@ -19,53 +26,76 @@ protected override void OnCreateManager(int capacity)
             view = worker.View;
 
             GenerateComponentGroups();
-
-            newlyAddedSpatialOSEntityComponentGroup = GetComponentGroup(typeof(NewlyAddedSpatialOSEntity));
-            onConnectedComponentGroup = GetComponentGroup(typeof(OnConnected));
-            onDisconnectedComponentGroup = GetComponentGroup(typeof(OnDisconnected));
         }
 
         private void GenerateComponentGroups()
         {
             foreach (var translationUnit in view.TranslationUnits.Values)
             {
-                translationUnit.CleanUpComponentGroups = new List<ComponentGroup>();
+                translationUnit.CleanUpComponentGroups = new Collections.List<ComponentGroup>();
                 foreach (ComponentType componentType in translationUnit.CleanUpComponentTypes)
                 {
                     translationUnit.CleanUpComponentGroups.Add(GetComponentGroup(componentType));
                 }
             }
+
+            MethodInfo addRemoveComponentActionMethod =
+                typeof(CleanReactiveComponentsSystem).GetMethod("AddRemoveComponentAction",
+                    BindingFlags.NonPublic | BindingFlags.Instance);
+
+            // Find all components with the RemoveAtEndOfTick attribute
+            foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
+            {
+                foreach (var type in assembly.GetTypes())
```

Comment:
LINQ ?

`assembly.GetTypes().Where((type) => ...).ForEach((type) => ...)`

Reply:
I personally think that makes this harder to read. But I will see if there is something I can do with this.